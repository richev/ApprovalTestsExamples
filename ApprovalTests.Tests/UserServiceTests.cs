using ApprovalTests;
using ApprovalTests.Reporters;
using ApprovalTests.Reporters.ContinuousIntegration;
using NUnit.Framework;

namespace Example.Services.Tests
{
    [UseReporter(typeof(NCrunchReporter))]
    [TestFixture]
    public class UserServiceTests
    {
        private UserService _userService;

        [SetUp]
        public void SetUp()
        {
            _userService = new UserService();
        }

        [Test]
        public void GetUsers_returns_expected_users()
        {
            var users = _userService.GetUsers();

            Approvals.Verify(Stringify.Items(users));
        }
    }
}
