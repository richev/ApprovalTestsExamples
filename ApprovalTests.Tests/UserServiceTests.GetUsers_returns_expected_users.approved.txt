Count = 2
[0].Key = {0bf6029f-31e5-467c-9a0f-e982482a0ede}
[0].Age = 20
[0].Name.First = 'Jack'
[0].Name.Last = 'Black'
[0].Addresses.Count = 2
[0].Addresses[0].Type = Business (2)
[0].Addresses[0].Street = 'Work Road'
[0].Addresses[0].City = 'Sydney'
[0].Addresses[0].State = 'NSW'
[0].Addresses[0].Postcode = '2000'
[0].Addresses[1].Type = Home (1)
[0].Addresses[1].Street = 'Home Road'
[0].Addresses[1].City = 'Fairlight'
[0].Addresses[1].State = 'NSW'
[0].Addresses[1].Postcode = '2094'
[0].SomeCodes.Count = 0
[1].Key = {cda51980-0b41-43db-9135-bf44471bd07f}
[1].Age = <null>
[1].Name.First = 'Jill'
[1].Name.Last = 'Blue'
[1].Addresses = <null>
[1].SomeCodes.Count = 3
[1].SomeCodes[0] = '01'
[1].SomeCodes[1] = '05'
[1].SomeCodes[2] = '11'
